src/transport/transport.ts

```diff
@@ -139,7 +139,7 @@ class Transport extends EventEmitter {
     }
 
     public registerMessageHandler(handler: MessageHandler): void {
-        this.messageHandlers.unshift(handler);
+        this.messageHandlers.push(handler);
```

Comment:
Is there a specific reason using 'unshift' before?

Reply:
The only one I can think of is that now if you wrap something in a listener and listen for the event you just got, you'll get it again in the new listener, but I'd consider that an improvement over current behavior. There may be more reasons though.